src/librustdoc/clean/inline.rs

```diff
@@ -421,7 +421,7 @@ crate fn build_impl(
                                 associated_trait.def_id,
                             )
                             .unwrap(); // SAFETY: For all impl items there exists trait item that has the same name.
-                        !tcx.get_attrs(trait_item.def_id).lists(sym::doc).has_word(sym::hidden)
```

Comment:
Are `lists` and `has_word` still used after this changed? You might be able to remove them altogether.

Reply:
Yes, they are still used, notably on `&[ast::Attribute]`.

Reply:
Gotcha - it might be a good follow up PR to make that use the same logic as `is_doc_hidden`, since apparently the current logic is very slow.